backend/pkg/api/connect/service/quota/v1/service.go

```diff
@@ -0,0 +1,169 @@
+	kafkaReq := s.kafkaClientMapper.listQuotasRequestToKafka(req.Msg)
+
+	resp, err := s.consoleSvc.DescribeClientQuotas(ctx, kafkaReq)
+	if err != nil {
+		return nil, connect.NewError(connect.CodeInternal, fmt.Errorf("failed to describe quotas: %w", err))
+	}
+
+	// Handle errors from the console service
+	if err = kerr.ErrorForCode(resp.ErrorCode); err != nil {
+		return nil, connect.NewError(connect.CodeInternal, fmt.Errorf("failed to describe quotas: %s", err))
+	}
+
+	// Map console response to protobuf response
+	quotaEntries, err := s.kafkaClientMapper.quotaItemsToProto(resp.Entries)
+	if err != nil {
+		return nil, connect.NewError(connect.CodeInternal, fmt.Errorf("failed to map quota items to proto: %w", err))
+	}
+
+	// Add pagination
+	var nextPageToken string
+	if req.Msg.GetPageSize() > 0 {
+		sort.SliceStable(quotaEntries, func(i, j int) bool {
+			if quotaEntries[i].Entity.EntityType != quotaEntries[j].Entity.EntityType {
+				return quotaEntries[i].Entity.EntityType < quotaEntries[j].Entity.EntityType
+			}
+			return quotaEntries[i].Entity.EntityName < quotaEntries[j].Entity.EntityName
+		})
+		page, token, err := pagination.SliceToPaginatedWithToken(quotaEntries, int(req.Msg.PageSize), req.Msg.GetPageToken(), "quotaEntry", func(x *v1.ListQuotasResponse_QuotaEntry) string {
+			return fmt.Sprintf("%d_%s", x.Entity.EntityType, x.Entity.EntityName)
+		})
+		if err != nil {
+			return nil, connect.NewError(
+				connect.CodeInternal,
+				fmt.Errorf("failed to apply pagination: %w", err),
+			)
+		}
+		quotaEntries = page
+		nextPageToken = token
+	}
+
+	response := &v1.ListQuotasResponse{
+		Quotas:    quotaEntries,
+		PageToken: nextPageToken,
+	}
+
+	return connect.NewResponse(response), nil
+}
+
+// BatchSetQuota creates client throughput quotas for specified entities.
+func (s *Service) BatchSetQuota(ctx context.Context, req *connect.Request[v1.BatchSetQuotaRequest]) (*connect.Response[v1.BatchSetQuotaResponse], error) {
+	kafkaReq, err := s.kafkaClientMapper.batchSetQuotaRequestToKafka(req.Msg)
+	if err != nil {
+		return nil, connect.NewError(connect.CodeInvalidArgument, err)
+	}
+
+	_, err = s.consoleSvc.AlterClientQuotas(ctx, kafkaReq)
+	if err != nil {
+		return nil, connect.NewError(connect.CodeInternal, fmt.Errorf("failed to create quota: %w", err))
+	}
+
+	response := &v1.BatchSetQuotaResponse{}
+
+	return connect.NewResponse(response), nil
+}
+
+// DeleteQuota deletes client quotas for specified entities and value types.
+func (s *Service) DeleteQuota(ctx context.Context, req *connect.Request[v1.DeleteQuotaRequest]) (*connect.Response[v1.DeleteQuotaResponse], error) {
+	kafkaReq, err := s.kafkaClientMapper.deleteQuotaRequestToKafka(req.Msg)
+	if err != nil {
+		return nil, connect.NewError(connect.CodeInvalidArgument, err)
+	}
+
+	_, err = s.consoleSvc.AlterClientQuotas(ctx, kafkaReq)
+	if err != nil {
+		return nil, connect.NewError(connect.CodeInternal, fmt.Errorf("failed to delete quota: %w", err))
+	}
```

Comment:
We need to inspect the inner kafka error here as well. You did it with the list quota rpc

See:

```go
type AlterClientQuotasResponseEntry struct {
	// ErrorCode is the error code for an alter on a matched entity.
	ErrorCode int16

	// ErrorMessage is an informative message if the alter on this entity failed.
	ErrorMessage *string
```